src/transformers/modeling_lxmert.py

```diff
@@ -0,0 +1,975 @@
+                    l_outputs[1:] if output_l_attentions else None,
+                    x_encoder_outputs[2:] if output_x_attentions else None,
+                ),
+            )
+
+        return LxmertModelOutput(
+            last_hidden_state_l=lang_output,
+            hidden_states_l=l_outputs[0][:-1],
+            attentions_l_encoder=output_l_attentions,
+            last_hidden_state_v=visn_output,
+            hidden_states_v=v_outputs[0][:-1],
+            attentions_v_encoder=output_v_attentions,
+            pooled_output_x_encoder=pooled_output,
+            attentions_x_encoder=output_x_attentions,
+        )
+
+
+@add_start_docstrings(
+    """Lxmert Model with a specified pre-training heads on top. """, LXMERT_START_DOCSTRING, LXMERT_INPUTS_DOCSTRING
+)
+class LxmertForPretraining(LxmertPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        # Configuration
+        self.config = config
+        self.num_answers = config.n_qa_labels
+        self.visual_loss_normalizer = config.visual_loss_normalizer
+
+        # Use of pre-training tasks
+        self.task_mask_lm = config.task_mask_lm
+        self.task_obj_predict = config.task_obj_predict
+        self.task_matched = config.task_matched
+        self.task_qa = config.task_qa
+
+        # Lxmert backbone
+        self.bert = LxmertModel(config)
+
+        # Pre-training heads
+        self.cls = LxmertPreTrainingHeads(config, self.bert.embeddings.word_embeddings.weight)
+        if self.task_obj_predict:
+            self.obj_predict_head = LxmertVisualObjHead(config)
+        if self.task_qa:
+            self.answer_head = LxmertVisualAnswerHead(config, self.num_answers)
+
+        # Weight initialization
+        self.init_weights()
+
+        # Loss functions
+        self.loss_fcts = {
+            "l2": SmoothL1Loss(reduction="none"),
+            "visn_ce": CrossEntropyLoss(ignore_index=-1, reduction="none"),
+            "ce": CrossEntropyLoss(ignore_index=-1),
+        }
+
+        visual_losses = {}
+        if config.visual_obj_loss:
+            visual_losses["obj"] = {"shape": (-1,), "num": config.n_object_labels, "loss": "visn_ce"}
+        if config.visual_attr_loss:
+            visual_losses["attr"] = {"shape": (-1,), "num": config.n_attr_labels, "loss": "visn_ce"}
+        if config.visual_obj_loss:
+            visual_losses["feat"] = {
+                "shape": (-1, config.visual_feat_dim),
+                "num": config.visual_feat_dim,
+                "loss": "l2",
+            }
+        self.visual_losses = visual_losses
+
+    def forward(
+        self,
+        input_ids,
+        visual_feats,
+        token_type_ids=None,
+        attention_mask=None,
+        masked_lm_labels=None,
+        obj_labels=None,
+        matched_label=None,
+        ans=None,
+        output_v_attentions=False,
+        output_l_attentions=False,
+        output_x_attentions=False,
```

Comment:
This is cool! In order to re-introduce the mixin we'll probably also have to add an argument `output_attentions` which would return all three.

Reply:
We probably should override the attention test directly in the `test_lxmert_modeling.py` file to check these directly instead, rather than implement an additional argument which wouldn't serve much purpose.

Reply:
hrm, I see. I actually think for the sake of simplicity, just returning the attention from each encoder would probably be easier for the end user. However, if you do like this implementation better, I think that changing the test directly, should not be that much of a problem!